### PR TITLE
 Fix missing predicates from the solved query graph when using greedy

### PR DESCRIPTION
**Pull request by davidegrohmann**
Also improved unnesting of Apply

### CODE REVIEW

**Comment by systay** [comment]
I don't quite get why we need to change `unnestApply` like that, @davidegrohmann. It's used from `LogicalPlanRewriter`, which is responsible for making sure all rewriters are called until no more changes can be done. See https://github.com/neo4j/neo4j/blob/3.0/community/cypher/cypher-compiler-3.0/src/main/scala/org/neo4j/cypher/internal/compiler/v3_0/planner/logical/plans/rewriter/LogicalPlanRewriter.scala#L26

**Comment by davidegrohmann** [comment]
@systay failing tests mostly, I'll revert that commit and try to fix the testing framework instead.

**Comment by davidegrohmann** [comment]
@systay updated

File: community/cypher/cypher/src/test/scala/org/neo4j/cypher/ShortestPathAcceptanceTest.scala [hunk 1]
```diff
@@ -210,4 +210,26 @@ class ShortestPathAcceptanceTest extends ExecutionEngineFunSuite with NewPlanner
...
+
+    val result = graph.execute(
+      """
+        |CYPHER PLANNER=COST
+        |MATCH p=(source:Person)-[:FRIEND_OF*]-(target:Person)
+        |WITH p WHERE length(p) > 1
+        |UNWIND nodes(p)[1..-1] as n
+        |RETURN n, count(*)""".stripMargin)
+
+    result.close()
```
Comment (Mats-SX) [inline community/cypher/cypher/src/test/scala/org/neo4j/cypher/ShortestPathAcceptanceTest.scala]: No assertions on the result?

Comment (davidegrohmann) [inline community/cypher/cypher/src/test/scala/org/neo4j/cypher/ShortestPathAcceptanceTest.scala]: No need! Joking I forgot, adding one.

**Pull request merged**